src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java

```diff
@@ -154,27 +153,31 @@ else if (modelingExercise.isExamExercise()) {
 
     /**
      * retrieves a modeling submission without assessment for the specified correction round and potentially locks the submission
-     *
+     * <p>
      * In case Compass is supported (and activated), this method also assigns a result with feedback suggestions to the submission
      *
      * @param lockSubmission whether the submission should be locked
      * @param correctionRound the correction round (0 = first correction, 1 = second correction
      * @param modelingExercise the modeling exercise for which a
      * @param isExamMode whether the exercise belongs to an exam
-     * @return a random modeling submission (potentially based on compass)
+     * @return a random modeling submission (potentially based on compass) if present
      */
-    public ModelingSubmission findRandomSubmissionWithoutExistingAssessment(boolean lockSubmission, int correctionRound, ModelingExercise modelingExercise, boolean isExamMode) {
-        var submissionWithoutResult = super.getRandomAssessableSubmission(modelingExercise, isExamMode, correctionRound)
-                .orElseThrow(() -> new EntityNotFoundException("Modeling submission for exercise " + modelingExercise.getId() + " could not be found"));
-        ModelingSubmission modelingSubmission = (ModelingSubmission) submissionWithoutResult;
+    public Optional<ModelingSubmission> findRandomSubmissionWithoutExistingAssessment(boolean lockSubmission, int correctionRound, ModelingExercise modelingExercise,
+            boolean isExamMode) {
+        var submissionWithoutResult = super.getRandomAssessableSubmission(modelingExercise, isExamMode, correctionRound);
+        if (submissionWithoutResult.isEmpty())
+            return Optional.empty();
+
+        ModelingSubmission modelingSubmission = (ModelingSubmission) submissionWithoutResult.get();
```

Comment:
This cast seems unnecessary. 
```suggestion
        ModelingSubmission modelingSubmission = submissionWithoutResult.get();
```

Reply:
`super.getRandomAssessableSubmission` returns a `Submission`, which is the super class of `ModelingSubmission`. With the method returning a `ModelingSubmission`, I might as well just cast it there.